### PR TITLE
Fixes #10569 ALC live map loading needs optimization

### LINKED ISSUE #10569: ALC live map loading needs optimization
## Bug description 

The visible area of the live map is not completely populated with adventure lab caches, itseems not optimized to cover the whole viewport in an adaptive way (just guessing)

### Steps to reproduce the problem

- Set global cache type filter to ALC (for easier view of the problem)
- Open live map and zoom out some steps

### Actual result after these steps 

ALCs are only shown in some central part of the map.
If you move the map another cluster around the center of the map gets shown.
Sometimes even some ALC vanish when new ones are loaded after slightly moving the map

### Expected result after these steps 

The map should show all available ALC in the shown viewport.
(Remark: Some limits might apply on low zoom as our internal cache cache / buffer has a certain maximum AFAIK)
So lets say:
The result should be comparable to what we do for "normal" caches on the live map. If the user sees some "normal caches" in an area of the map he assumes, that this is all. But in fact some ALC might be missing.

## c:geo version 

2021.05.08-NB

## Reproducible 

Yes

## Additional context 

IMHO this should not be fixed in a hurry, but carefully compared to what we do with GC:
We want to avoid to many server requests but still assure that we show reliable information...thats sometimes difficult. 
@moving-bits 
You optimized quite a lot for the live map in the past. Maybe you know how to make ALC behave in a comparable way.

### PR DESCRIPTION
**Pull request by baiti**
searchByBBox() now implements a radius calculation that allows to fill a rectangular map area. See discussion: #10569

### CODE REVIEW

**Comment by baiti** [comment]
ohh  nooo ... 165 files changed! git again.

**Comment by baiti** [comment]
> ohh nooo ... 165 files changed! git again.

I guess I fixed it .... puuh. I was about to give that git challenge award back to @fm-sys

**Comment by fm-sys** [comment]
Sorry! I had not thought of it, but we also have a method in the `Geopoint` class which does the calculation for us and makes things way easier:
`oneGeopoint.distanceTo(otherGeopoint)`

**Comment by baiti** [comment]
> Sorry! I had not thought of it, but we also have a method in the `Geopoint` class which does the calculation for us and makes things way easier:
> `oneGeopoint.distanceTo(otherGeopoint)`

Oooook .... that really simplifies the code a lot. Will do tonight.

**Comment by baiti** [comment]
@Lineflyer I'd consider it ready for a merge. At least from a coding perspective the diff should now be reasonably simple.
I have done a sniff test on the emulator and I believe it is "ok" now, however, it would certainly be valuable if you did a quick test too.

**Comment by Lineflyer** [comment]
Sure, I will merge it tonight and merge upstream to test in nightlies before updating the beta version out there next days.

**Comment by baiti** [comment]
Don't merge yet. The new "toDistance" function seems to get "km", I need "m" .... factor of 1000 needs to be applied.

**Comment by baiti** [comment]
With commit https://github.com/cgeo/cgeo/pull/10577/commits/39a398378a2f7a08bf19b91ea1d642d5e9fc2e8e I just did, it should now finally be working.

File: main/src/cgeo/geocaching/connector/lc/LCApi.java [hunk 1]
```diff
@@ -67,10 +67,27 @@ protected static Geocache searchByGeocode(final String geocode) {
...
+        final double lon2 = viewport.getLongitudeMin();
+        final double latcenter = (lat1 + lat2) / 2;
+        final double loncenter = (lon1 + lon2) / 2;
+        final Geopoint gp1 = new Geopoint(lat1, lon1);
+        final Geopoint gp2 = new Geopoint(lat2, lon2);
+        final double radius = gp1.distanceTo(gp2) * 500; // we get diameter in km, need radius in m
+        Log.d("_LC Radius: " + String.valueOf((int) radius));
+        final Parameters params = new Parameters("skip", "0");
+        params.add("take", "500");
+        params.add("radiusMeters", String.valueOf((int) radius);
```
Comment (fm-sys) [inline main/src/cgeo/geocaching/connector/lc/LCApi.java:82]: A ")" is missing here...

Comment (baiti) [reply main/src/cgeo/geocaching/connector/lc/LCApi.java]: that happens if you're not at the android studio system and commit changes without doing a compile first.

I need to get my commandline gradle comand working. Perhaps that is the opportunity now to learn it. How do I trigger a build from the commandline?

**Comment by Lineflyer** [comment]
Thanks, lets try it out :)

File: main/src/cgeo/geocaching/connector/lc/LCApi.java (hunk 1 above)
Comment (fm-sys) [reply main/src/cgeo/geocaching/connector/lc/LCApi.java]: I think a `./gradlew build` should do the trick. But cannot say as I don't use it myself...

**Pull request merged**